libexec/rtld-elf/aarch64/rtld_c18n_asm.S

```diff
@@ -400,6 +400,13 @@ TRAMP(tramp_update_fp_untagged)
 	clrtag	c29, TRUSTED_STACK_C
 TRAMPEND(tramp_update_fp_untagged)
 
+TRAMP(tramp_count_entry)
+1:	ldr	c24, #0		/* To be patched at runtime */
+	stadd	w25, [c24]
```

Comment:
Where does this w25 come from?

Reply:
From the `gctag	x25, c30` above and we can assume that it is always 1. This is done to save instructions and assumes that the program is well-behaved.

Reply:
Firstly, these things *have* to be commented. There are so many of these implicit dependencies floating around within and across trampoline fragments that it's really hard to keep track of them all, and it's just asking for someone to break this in future by using a different register for the GCTAG (which would be reasonable, because who would expect it to be used many instructions later?). Secondly, having an untagged return capability seems a totally valid thing to do if you're making a call to a function that won't return. I don't think you can currently easily end up with it from a compiler, but I see no reason why it wouldn't be valid.